test/server/ip-tools.js

```diff
@@ -33,6 +33,27 @@ describe("IP tools", () => {
 		assert.deepEqual(cidrRange, stringRange);
 	});
 
+	it('should not parse invalid ranges', () => {
+		assert.equal(IPTools.stringToRange('42.42.10.0 - 42.42.5.0'), null);
+		assert.equal(IPTools.stringToRange('250.0.0.0 - 260.0.0.0'), null);
+		assert.equal(IPTools.stringToRange('250.0.0.0/43'), null);
+		assert.equal(IPTools.stringToRange('250.0.0.0/16x'), null);
+		assert.equal(IPTools.stringToRange('250.0.0.0.1/24'), null);
+		assert.equal(IPTools.stringToRange('250.0/24'), null);
+		assert.equal(IPTools.stringToRange('250.0.0.0.1 - 250.0.0.2'), null);
+		assert.equal(IPTools.stringToRange('250.0.0.0.*'), null);
+		assert.equal(IPTools.stringToRange('250.0.0'), null);
+	});
+
```

Comment:
```suggestion

	it('should correctly parse shorthand IPs', () => {
		assert.equal(IPTools.ipToNumber('127.0.0.1'), IPTools.ipToNumber('127.1'));
	});

```

Reply:
It's somewhat questionable what we should do here. Rejecting them for not being valid IPs is also plausibly valid.

Reply:
I'm not really familiar with IP shorthand, but I don't think it would be used internally by PS. The question is whether it's worth supporting for user-facing commands (if so, the `ipRegex` should probably be updated to support it).

Reply:
It's probably not worth supporting. I've never seen anyone use it in my life; I just know of its existence because it used to be a fun way to get around internet filters - if an IP is banned in one format, just try another format!

Reply:
lmao, sounds like fun, but not something we need to support